### PR TITLE
Theta bidask

### PR DESCRIPTION
**Pull request by haochili**
1. Enabled bid/ask feature with thetadata 
2. Added 36 unit testing feature for thetadata_helper.py
3. Won't show thetadata download on weekends

### What change is being made?

Added support for fetching and handling bid/ask quote data from ThetaData, including new methods, updated caching mechanisms, and comprehensive unit tests.

### Why are these changes being made?

These changes enhance the backtesting capabilities by incorporating bid/ask quote data, which is crucial for more accurate and realistic simulations. This update also includes improvements in data handling, caching, and error management to ensure robustness and reliability.

### CODE REVIEW

File: lumibot/backtesting/thetadata_backtesting.py [hunk 1]
```diff
@@ -128,20 +145,35 @@ def update_pandas_data(self, asset, quote, length, timestep, start_dt=None):
...
+                self.datetime_end,
+                timespan=ts_unit,
+                quote_asset=quote_asset,
+                dt=date_time_now,
+                datastyle="quote"
+            )
+            df = pd.concat([df_ohlc, df_quote], axis=1, join='inner')
             # save df to csv file
-            # df.to_csv(f"theta_csv/{date_time_now}_{asset.strike}_{asset.expiration}_{asset.right}.csv")
+            # if asset.asset_type == "option":
```
Comment (grzesir) [inline lumibot/backtesting/thetadata_backtesting.py:175]: clean up comments

File: lumibot/tools/thetadata_helper.py [hunk 2]
```diff
@@ -305,23 +271,63 @@ def update_df(df_all, result):
...
+            else:
+                df_all = df_all.set_index("datetime").sort_index()
+
+            # convert df_all index to UTC if not already
+            if not df.index.tzinfo:
+                df_all.index = df_all.index.tz_localize(ny_tz).tz_convert(pytz.utc)
+            else:
+                df_all.index = df_all.index.tz_convert(pytz.utc)
+
+        # df.index = df.index.tz_localize("UTC") if df.index.tzinfo is None else df.index
```
Comment (grzesir) [inline lumibot/tools/thetadata_helper.py:322]: comments cleanup

**Review by grzesir — CHANGES_REQUESTED** [review]

**Review by grzesir — APPROVED** [review]

File: lumibot/backtesting/thetadata_backtesting.py (hunk 1 above)
Comment (haochili) [reply lumibot/backtesting/thetadata_backtesting.py]: I cleaned up the comments

File: lumibot/tools/thetadata_helper.py (hunk 2 above)
Comment (haochili) [reply lumibot/tools/thetadata_helper.py]: This is cleaned up too

**Pull request merged**